### PR TITLE
Changes to set default driver for running properites computation

### PR DESCRIPTION
**Pull request by ahurta92**
## Description
The purpose of the PR is to be able to set a Dataset's default driver to something other than the default "energy".  I was hoping to compute dipole moments with nwchem on my Dataset.  In order to do so, I need to use the "properties" driver.  These changes allow me to do that.  It also allows get_values to return energy values when the driver is "properties".  As a note, I wasn't able to set the default driver using ds.data.default_driver="properties" because the Datamodel class is immutable.

## Changelog description
Creates set_default_driver() function to be able to set the default driver on Dataset collections.  

## Status

- [ ] Code base linted
- [ ] Ready to go

### CODE REVIEW

File: qcfractal/interface/collections/dataset.py [hunk 1]
```diff
@@ -700,7 +700,7 @@ def _get_native_values(
         DataFrame
             A DataFrame of the queried parameters
         """
-        au_units = {"energy": "hartree", "gradient": "hartree/bohr", "hessian": "hartree/bohr**2"}
+        au_units = {"energy": "hartree", "gradient": "hartree/bohr", "hessian": "hartree/bohr**2","properties","hartree"}
```
Comment (dgasmith) [inline qcfractal/interface/collections/dataset.py]: Needs to be `"properties": "hartree"`

Comment (ahurta92) [reply qcfractal/interface/collections/dataset.py]: Yeah sure!  I'm guessing you want me to add the test to test_dataset.py

**Comment by dgasmith** [comment]
Yes, please!

**Comment by bennybp** [comment]
Datasets have radically changed in v0.50, and most defaults are no longer settable in a singlepoint dataset. I would be happy to talk and go over the new interface if you would like!

**Pull request closed without merging**